src/editor/nodes/AudioZoneNode.js

```diff
@@ -0,0 +1,125 @@
+import { Material, BoxBufferGeometry, Mesh, BoxHelper } from "three";
+import AudioParams from "../objects/AudioParams";
+import EditorNodeMixin from "./EditorNodeMixin";
+
+const requiredProperties = ["enabled", "inOut", "outIn"];
+
+export default class AudioZoneNode extends EditorNodeMixin(AudioParams) {
+  static componentName = "audio-zone";
+
+  static nodeName = "Audio Zone";
+
+  static _geometry = new BoxBufferGeometry();
+
+  static _material = new Material();
+
+  static async deserialize(editor, json) {
+    const node = await super.deserialize(editor, json);
+
+    const zoneProps = json.components.find(c => c.name === "audio-zone").props;
+
+    node.enabled = zoneProps.enabled;
+    node.inOut = zoneProps.inOut;
+    node.outIn = zoneProps.outIn;
+
+    const audioParamsProps = json.components.find(c => c.name === "audio-params").props;
+
+    node.audioType = audioParamsProps.audioType;
+    node.gain = audioParamsProps.gain;
+    node.distanceModel = audioParamsProps.distanceModel;
+    node.rolloffFactor = audioParamsProps.rolloffFactor;
+    node.refDistance = audioParamsProps.refDistance;
+    node.maxDistance = audioParamsProps.maxDistance;
+    node.coneInnerAngle = audioParamsProps.coneInnerAngle;
+    node.coneOuterAngle = audioParamsProps.coneOuterAngle;
+    node.coneOuterGain = audioParamsProps.coneOuterGain;
+
+    return node;
+  }
+
+  constructor(editor) {
+    super(editor, editor.audioListener);
+
+    const boxMesh = new Mesh(AudioZoneNode._geometry, AudioZoneNode._material);
+    const box = new BoxHelper(boxMesh, 0x00ff00);
+    box.layers.set(1);
+    this.helper = box;
+    this.add(box);
+    this.enabled = true;
+    this.inOut = true;
+    this.outIn = true;
+  }
+
+  copy(source, recursive = true) {
+    if (recursive) {
+      this.remove(this.helper);
+    }
+
+    super.copy(source, recursive);
+
+    if (recursive) {
+      const helperIndex = source.children.indexOf(source.helper);
+
+      if (helperIndex !== -1) {
+        this.helper = this.children[helperIndex];
+      }
+    }
+
+    this.enabled = source.enabled;
+    this.inOut = source.inOut;
+    this.outIn = source.outIn;
+
+    return this;
+  }
+
+  serialize() {
+    return super.serialize({
+      "audio-zone": {
+        enabled: this.enabled,
+        inOut: this.inOut,
+        outIn: this.outIn
+      },
+      "audio-params": {
+        audioType: this.audioType,
+        gain: this.gain,
+        distanceModel: this.distanceModel,
+        rolloffFactor: this.rolloffFactor,
+        refDistance: this.refDistance,
+        maxDistance: this.maxDistance,
+        coneInnerAngle: this.coneInnerAngle,
+        coneOuterAngle: this.coneOuterAngle,
+        coneOuterGain: this.coneOuterGain
+      }
+    });
+  }
+
+  prepareForExport() {
```

Comment:
So.. This looks like what we discussed we wanted yesterday, where no mesh would actually get exported, but I am a bit surprised how this was working currently since the hubs-side code wanted the object's bounding box...

Reply:
The are applying the world transform to the BBAA in the client here:
https://github.com/mozilla/hubs/blob/ad8e269ceaac57e28b10cb781c755a74ac0a1df8/src/components/audio-zone.js#L68-L73 
```
  getBoundingBox: (function() {
    const bbaa = new THREE.Box3();
    return function() {
      return bbaa.copy(this.debugMesh.geometry.boundingBox).applyMatrix4(this.el.object3D.matrixWorld);
    };
  })(),
```
We are also adding the debug mesh to the root object node so the transform is applied to the child mesh and the transform is the exported GLTF so the debug object has the right transform.

Reply:
Thanks!!! amazing work!